### PR TITLE
SDK - Using Airflow ops in Pipelines

### PR DESCRIPTION
**Pull request by Ark-kun**
This CL adds support for converting arbitrary Airflow operators to Pipelines components so that it can be used as a pipeline task.

```python
from airflow.operators.bash_operator import BashOperator
bash_op = kfp.components.create_component_from_airflow_op(BashOperator)

@pipeline()
def my_pipeline():
  bash_task = bash_op('echo "Hello from $SHELL"', xcom_push=True)
```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1483)

### CODE REVIEW

File: sdk/python/kfp/components/_airflow_op.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+from typing import List
+
+from ._python_op import _func_to_component_spec, _create_task_factory_from_component_spec
+
+
+_default_airflow_base_image = 'apache/airflow@sha256:7f60cbef6bf92b1f3a5b4e46044911ced39736a8c3858284d3c5a961b3ba8735'
+
+
+def create_component_from_airflow_op(op_class, base_image=_default_airflow_base_image, result_output_name='Result', variable_output_names=None, xcom_output_names=None, modules_to_capture: List[str] = None):
```
Comment (hongye-sun) [inline sdk/python/kfp/components/_airflow_op.py]: Could you add a comment about this method?

Comment (hongye-sun) [inline sdk/python/kfp/components/_airflow_op.py]: Could you add an example on how to use it?

Comment (Ark-kun) [reply sdk/python/kfp/components/_airflow_op.py]: The description of this PR has an example, but I can add one in the code.

Comment (hongye-sun) [reply sdk/python/kfp/components/_airflow_op.py]: A basic sample will also increase the visibility of the feature. 
Do you know if the code is ready to ship? any known limitation?

Comment (Ark-kun) [reply sdk/python/kfp/components/_airflow_op.py]: > Do you know if the code is ready to ship? any known limitation?

This PR depended on https://github.com/kubeflow/pipelines/pull/1435 https://github.com/kubeflow/pipelines/pull/1511 and is also affected by https://github.com/kubeflow/pipelines/issues/1485, but the biggest issue for now is https://github.com/kubeflow/pipelines/issues/1507 .

The new notebooks are running python 3.6. Fortunately, `airflow` images also run `3.6`, so this PR can probably move forward soon. (Bad news is that `tensorflow` images are `3.5`)

Comment (hongye-sun) [reply sdk/python/kfp/components/_airflow_op.py]: sg. thanks for the info.

Comment (Ark-kun) [reply sdk/python/kfp/components/_airflow_op.py]: I'll add the docstring, thanks.

**Comment by animeshsingh** [comment]
@Ark-kun is this only for bash operator? Other operators being supported as well?  Also can this be extended to take an overall Airflow DAG, and execute as one of the 'steps' in KFP, assuming an Airflow deployment exists and KFP is made aware of it?

**Comment by animeshsingh** [comment]
@paveldournov you have a viewpoint on the above question? cc @vicaire @jlewi

**Comment by Ark-kun** [comment]
> @Ark-kun is this only for bash operator? Other operators being supported as well?

This feature supports all Airflow operators.

>can this be extended to take an overall Airflow DAG, and execute as one of the 'steps' in KFP

It could. How do you envision the API for AirFlow DAGs?

>assuming an Airflow deployment exists and KFP is made aware of it?

This might be unnecessary

File: sdk/python/kfp/components/_airflow_op.py (hunk 1 above)
Comment (Ark-kun) [reply sdk/python/kfp/components/_airflow_op.py]: Done

**Comment by animeshsingh** [comment]
> It could. How do you envision the API for AirFlow DAGs?

Two modes 
1. maybe a component can take Airflow DAG as an input, and calls Airflow deployment
2. Pipeline DSL can compile to airflow DAG?

**Comment by NieRonghua** [comment]
> > @Ark-kun is this only for bash operator? Other operators being supported as well?
> 
> This feature supports all Airflow operators.

Hi, I’m an Airflow user, now i want to create components by using create_component_from_airflow_op(SimpleHttpOperator), but i do not konw how to define the airflow‘s conn such as msyql conn or http conn, can u give me an example? thx

**Pull request merged**